PowerShell/CipherTrustManager/ConnectionsMgr/CipherTrustManager-ConnectionMgr-AWS.psm1

```diff
@@ -0,0 +1,1197 @@
+    .PARAMETER description
+        (Optional) Description of the connection.
+    .PARAMETER is_role_anywhere
+        (Optional) Use this switch to create connections of type AWS IAM Anywhere with temporary credentials.
+    .PARAMETER anywhere_role_arn
+        (Required if using IAM Role Anywhere) Specify AWS IAM Anywhere Role ARN.
+    .PARAMETER anywhere_role_cert
+        (Required if using IAM Role Anywhere) Upload the PEM-formatted certificate text for AWS IAM Anywhere Cloud connections. 
+        While it can be used from the command-line, the switch is best used when running automation scripts. Populate a variable with the PEM-formatted certificate then pass the variable to the command.
+    .PARAMETER anywhere_role_certfile
+        (Required if using IAM Role Anywhere) Specify the filename for a PEM certificate for AWS IAM Anywhere Cloud connections.
+    .PARAMETER anywhere_role_privkey
+        (Required if using IAM Role Anywhere) Upload the PEM-formatted private key text for AWS IAM Anywhere Cloud connections. 
+        While it can be used from the command-line, the switch is best used when running automation scripts. Populate a variable with the PEM-formatted private key then pass the variable to the command.
+    .PARAMETER anywhere_role_keyfile
+        (Required if using IAM Role Anywhere) Specify the filename for a PEM private key for AWS IAM Anywhere Cloud connections.
+    .PARAMETER anywhere_profile_arn
+        (Required if using IAM Role Anywhere) Specify AWS IAM Anywhere Profile ARN.
+    .PARAMETER anywhere_trust_anchor_arn
+        (Required if using IAM Role Anywhere) Specify AWS IAM Anywhere Trust Anchor ARN.
+    .PARAMETER metadata
+        (Optional) Optional end-user or service data stored with the connection. Use key/value pairs separated by a semi-colon. Can be a comma-separated list of metadata pairs. 
+        e.g. -metadata "red:stop,green:go,blue:ocean"
+    .EXAMPLE
+        PS> New-CMAWSConnection -name MyTestAWSConnection2 -description "This is my Test AWS Connection" -access_key_id abc123abc123 -secret_access_key xyz987xyz987 -assume_role_arn "arn:aws:iam::123456789012:user/johndoe" -assume_role_external_id EXT_ROLE_ID -aws_region us-west-1 -aws_sts_regional_endpoints regional -cloud_name aws-us-gov -metadata "red:stop,:green:go,blue:ocean" 
+    .EXAMPLE
+        PS> New-CMAWSConnection -name MyTestAWSConnection3 -description "This is my Test AWS Connection" -assume_role_arn "arn:aws:iam::123456789012:user/johndoe" -assume_role_external_id EXT_ROLE_ID -aws_region us-west-1 -aws_sts_regional_endpoints regional -cloud_name aws-us-gov -is_role_anywhere -anywhere_role_arn "arn:aws:iam::123456789012:user/johndoe" -anywhere_role_certfile mongocert.pem -anywhere_role_keyfile mongokey.pem -anywhere_profile_arn "arn:aws:iam::123456789012:user" -anywhere_trust_anchor_arn "arn:aws:iam::123456789012:user/johndoe" -metadata "red:stop,green:go,blue:ocean"
```

Comment:
someday, we would love to add examples of how you can do the same with secrets in CSM instead of directly passing them as string-based args... someday :)

Reply:
that would definitely be cool